src/App.tsx

```diff
@@ -15,71 +16,122 @@ export const goods = [
-            >
-              +
-            </button>
-          </td>
+type State = {
+  selectedGood: string,
+};
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  handleAdd = (good: string) => {
+    this.setState(
+      {
+        selectedGood: good,
+      },
+    );
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  handleRemove = () => {
+    this.setState(
+      {
+        selectedGood: '',
+      },
+    );
+  };
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+    return (
+      <main className="section container">
+        {
+          selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
+                <GoodsButton
+                  dataCy="ClearButton"
+                  content=""
+                  classes={['delete', 'ml-3']}
+                  handler={this.handleRemove}
+                />
+              </h1>
+            )
+            : (<h1 className="title">No goods selected</h1>)
+        }
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={classNames(
+                  {
+                    'has-background-success-light': good === selectedGood,
+                  },
+                )}
+              >
+                <td>
+                  {
+                    good === selectedGood
+                      ? (
+                        <GoodsButton
+                          dataCy="RemoveButton"
+                          content="-"
+                          classes={['button', 'is-info']}
+                          handler={this.handleRemove}
+                        />
+                      )
+                      : (
+                        <GoodsButton
+                          dataCy="AddButton"
+                          content="+"
+                          handler={() => this.handleAdd(good)}
+                        />
```

Comment:
```suggestion
                        <GoodsButton
                          dataCy="AddButton"
                          content="+"
                          type="button"
                          classes={['button']}
                          handler={() => this.handleAdd(good)}
                        />
```
If you use classes library here you also need  add class to second button and type.